github/MainClass.py

```diff
@@ -808,8 +808,12 @@ def get_app(self, slug=github.GithubObject.NotSet):
         if slug is github.GithubObject.NotSet:
             # with no slug given, calling /app returns the authenticated app,
             # including the actual /apps/{slug}
-            headers, data = self.__requester.requestJsonAndCheck("GET", "/app")
-            return GithubApp.GithubApp(self.__requester, headers, data, completed=True)
+            warnings.warn(
+                "Argument slug is mandatory, calling this method without the slug argument is deprecated, please use "
+                "github.GithubIntegration(auth=github.Auth.AppAuth(...)).get_app() instead",
+                category=DeprecationWarning,
+            )
+            return GithubIntegration(auth=self.__requester._Requester__auth).get_app()
```

Comment:
Please rebase and access `Requester.auth`:
```suggestion
            return GithubIntegration(auth=self.__requester.auth).get_app()
```